### PR TITLE
[WIP] PR to enable features on OOP on by default

### PR DESCRIPTION
**Pull request by heejaechang**
removed all options no longer needed.

user still has a way to run all these in proc by turning off OOP

### CODE REVIEW

**Comment by sharwell** [comment]
Why not using one of the two pull requests I sent for this?

File: src/Features/Core/Portable/IncrementalCaches/SymbolTreeInfoIncrementalAnalyzerProvider.cs [hunk 1]
```diff
@@ -198,8 +196,10 @@ private async Task UpdateSymbolTreeInfoAsync(Project project, CancellationToken
                     return;
                 }
 
-                if (!RemoteFeatureOptions.ShouldComputeIndex(project.Solution.Workspace))
+                var client = await project.Solution.Workspace.TryGetRemoteHostClientAsync(cancellationToken).ConfigureAwait(false);
+                if (client != null)
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/IncrementalCaches/SymbolTreeInfoIncrementalAnalyzerProvider.cs]: this feels like a very weird way to do things.  Can we just have a bit saying if we're the local-editing-workspace, vs the global-searching-workspace?

**Comment by heejaechang** [comment]
@sharwell ah, we already had? didn't know. if there is already one, I can close this.

File: src/Features/Core/Portable/IncrementalCaches/SymbolTreeInfoIncrementalAnalyzerProvider.cs (hunk 1 above)
Comment (heejaechang) [reply src/Features/Core/Portable/IncrementalCaches/SymbolTreeInfoIncrementalAnalyzerProvider.cs]: I thought I am following the pattern we already have. all other ones basically doing the same thing. if there is no remote host, it does work inproc, if there is, it call the remote to do the work.

here point is using remote host existance to figure out where work should be done.

**Comment by heejaechang** [comment]
closing. looks like @sharwell already have PR to enable these.

**Comment by sharwell** [comment]
@heejaechang I implemented two different approaches. We can discuss the approaches in Monday's design meeting and pick the one we want to use.

**Comment by heejaechang** [comment]
@sharwell okay. not sure what approach it needs except deleting all this RemoteFeatureOptions? since once we decide to run this in OOP, those options are not needed.

**Pull request closed without merging**